### DISCUSSION TITLE
Project Intro

### DISCUSSION THREAD

**Discussion by wiverson** [supabase/gotrue-csharp#64] [General]
@acupofjose  I was going to put this in a comment for the last PR, but realized that was dumb and so I'm putting it here. :)

Bit of background - I'm building a mobile client in Unity targeting iOS and Android. My background (for a very, very [long time)](https://www.linkedin.com/in/wiverson/) is mainly in Java at scale. A few years ago I shipped [BlazeSky](https://store.steampowered.com/app/1208980/BlazeSky/) in Unity as the solo dev. I'm totally mixing it up this time, using Supabase as the backend and seeing if I can get away w/o a middle-tier. Not used to PL/pgSQL but I'm coming up to speed fast, using that with PostREST is pretty dang cool. Hilariously ChatGPT is proving quite helpful for writing the PL/pqSQL stuff. Liking that better/finding that suits my background better than trying to get it all working via RLS/GraphQL.

It's not announced or anything yet, but the app is an allowance/chores app for families. Expecting to have a Unity parent app, a child app, and probably an Apple Watch app as well. Unity is giving me C#, fast native builds, lots of animation/effects options (esp for the child app). The Apple Watch app will likely just hit a handful of REST services directly with no Supabase client. I expect the combo of Swift + Apple Watch running in the simulator will be interesting.

Right now I'm busy writing test cases to get this stuff all to as Rock Solid(tm) as I can esp around the auth stuff. I'm sure there will be a few more minor PRs incoming as I deal with stuff like getting email confirm rock solid, some doc updates, etc. I'm sure some of that will emerge as I do physical device testing.

I am planning on making YouTube videos on the dev side of things, right now it's just getting it all working. It's a pity some of this stuff is a real pain to write test cases for (esp a bunch of the failure mode stuff) but so far it's all going quite well.

Thank you for writing/managing the supabase-csharp stuff - it's absolutely making all of this stuff a delight.

**Comment by acupofjose** [comment]
I've wondered about Unity being used beyond game development and instead for applications! Very cool. Blaze Sky looks like it was quite the undertaking, and it shows. The scale of the world/game looks sizeable, especially for a solo dev. 

Looking forward to what kind of dev videos you put out for the project!

**Reply by wiverson** [reply]
It certainly works. I think my two picks for x-plat dev right now are SvelteKit and Unity. SvelteKit + Capacitor works pretty well but as soon as you get to anything native it's hand-writing Swift & Android/Kotlin/etc. I need sign in with Apple and RevenueCat and it needs to be rock solid.

Unity has the benefit of a huge install base and a massive asset store. Plus the foreign function stuff in C#/Unity is much nicer/easier IMHO if you do have to bag out to native for something, but TBH you rarely have to do that as it's much more likely that someone already knocked out an implementation already.

Weird pro/con is the Unity UI builder tool. I have gotten to the point where I vastly prefer using a UI builder as otherwise building a responsive UI is a nightmare. So, with Unity I get a UI builder, plus particle effects, 3d, etc etc etc as well as perfectly nice ordinary UI stuff. All on top of C# and with nice tooling to target virtually every device out there (well, not Apple Watch but what can you do, lol).
